File: src/main/java/arcana/common/items/ModItemTags.java
package arcana.common.items;

import net.minecraft.item.Item;
import net.minecraft.tags.ItemTags;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.common.Tags;

import static arcana.utils.Util.arcLoc;

public class ModItemTags {
    public static Tags.IOptionalNamedTag<Item> SCRIBING_TOOLS = arcTag("scribing_tools");

    public static Tags.IOptionalNamedTag<Item> arcTag(String name){
        return ItemTags.createOptional(arcLoc(name));
    }
}

File: src/main/kotlin/arcana/common/items/ModItemTags.kt
package arcana.common.items

import arcana.utils.Util
import net.minecraft.item.Item
import net.minecraft.tags.ItemTags
import net.minecraftforge.common.Tags.IOptionalNamedTag

object ModItemTags {
    var SCRIBING_TOOLS: IOptionalNamedTag<Item> = arcTag("scribing_tools")

    private fun arcTag(name: String) = ItemTags.createOptional(Util.arcLoc(name))
}